prow/config/config.go

```diff
@@ -668,6 +668,14 @@ func (c *Config) finalizeJobConfig() error {
 			return errors.New("no default GCS credentials secret provided for plank")
 		}
 
+		if c.Plank.DefaultDecorationConfig.Timeout == nil {
```

Comment:
I am not sure this is sufficient -- the issue is, this is for job configuration coming from the YAML on disk, but we want to make sure that when we read the values from a ProwJob CRD in etcd we get no nil pointers. I was thinking we have a 

```go
func (c *DecorationConfig) Timeout() Duration {
    if c.Timeout == nil {
        return &Duration{0}
    }
    return c.Timeout
}
```

And use that in the consumres

Reply:
addressed